File: in_Python/1324 Print Words Vertically.py
class Solution(object):
    def printVertically(self, s):
        """
        :type s: str
        :rtype: List[str]
        """
        temp = []
        maxlen = 0
        for i in s.split(' '):
            if len(i) > maxlen:
                maxlen = len(i)
        t = s.split(' ')
        n = len(t)
        # temp shall be a maxlen-by-n matrix
        for i in range(maxlen):
            row = []
            for j in range(n):
                row.append(' ')
            temp.append(row)
            
        for idx in range(n):
            ss = t[idx]
            for r in range(len(ss)):
                temp[r][idx] = ss[r]
        res = []
        for row in range(maxlen):
            cur = temp[row]
            st = ""
            for c in cur:
                st += c
            res.append(st.rstrip())
        return res

File: in_Java/1324 Print Words Vertically.java
class Solution {
    public List<String> printVertically(String s) {
        String[] str = s.split(" ");
        int maxlen = 0;
        for(String x : str){
            if(maxlen < x.length()) maxlen = x.length();
        }
        char[][] t = new char[str.length][maxlen];
        for(int i = 0; i < str.length; i++){
            for(int j = 0; j < maxlen; j++){
                t[i][j] = ' ';
            }
        }
        for(int i = 0; i < str.length; i++){
            String cur = str[i];
            for(int j = 0; j < cur.length(); j++){
                t[i][j] = cur.charAt(j);
            }
        }
        List<String> res = new ArrayList<>();
        String temp;
        for(int c = 0; c < maxlen; c++){
            temp = "";
            for(int r = 0; r < str.length; r++){
                if(t[r][c] == ' ') temp += " ";
                else temp += String.valueOf(t[r][c]);
            }
            res.add(removeT(temp));
        }
        return res;
    }
    private String removeT(String s){
        int n = s.length()-1;
        while(s.charAt(n) == ' '){
            n--;
        }
        return s.substring(0,n+1);
    }
}